### PR TITLE
Add liberty support [DONE]

### PR DESCRIPTION
**Pull request by epcim**
Liberty update

- add ring alias
- customize ring/policies configuration
- tempauth support
- ...
- fixes

### CODE REVIEW

File: swift/account.sls [hunk 1]
```diff
@@ -6,6 +6,16 @@ swift_account_packages:
   pkg.installed:
   - names: {{ account.pkgs }}
 
+swift_account_node_directory:
+  file.directory:
+  - name: /srv/node
```
Comment (fpytloun) [inline swift/account.sls]: Hard-coded directory, make this a parameter

File: swift/common.sls [hunk 2]
```diff
@@ -25,8 +26,18 @@ swift_common_packages:
   file.managed:
   - source: salt://swift/files/{{ common.version }}/memcache.conf
   - template: jinja
-  - user: root
-  - group: root
+  - user: swift
```
Comment (fpytloun) [inline swift/common.sls]: root:swift should be enough. Swift should be happy with read perms.

File: swift/common.sls [hunk 3]
```diff
@@ -6,17 +6,18 @@ swift_common_packages:
 
 /etc/swift:
   file.directory:
-  - user: root
-  - group: root
+  - user: swift
```
Comment (fpytloun) [inline swift/common.sls]: root:swift should be enough, swift should be happy with rx perms

File: swift/common.sls [hunk 4]
```diff
@@ -6,17 +6,18 @@ swift_common_packages:
...
   - require:
     - pkg: swift_common_packages
 
 /etc/swift/swift.conf:
   file.managed:
   - source: salt://swift/files/{{ common.version }}/swift.conf
   - template: jinja
-  - user: root
-  - group: root
+  - user: swift
```
Comment (fpytloun) [inline swift/common.sls]: root:swift should be enough

File: swift/common.sls [hunk 5]
```diff
@@ -25,8 +26,18 @@ swift_common_packages:
...
+  - group: swift
+  - mode: 644
+  - require:
+    - file: /etc/swift
+
+/etc/swift/dispersion.conf:
+  file.managed:
+  - source: salt://swift/files/{{ common.version }}/dispersion.conf
+  - template: jinja
+  - user: swift
```
Comment (fpytloun) [inline swift/common.sls]: same here

File: swift/container.sls [hunk 6]
```diff
@@ -6,6 +6,16 @@ swift_container_packages:
   pkg.installed:
   - names: {{ container.pkgs }}
 
+swift_container_node_directory:
+  file.directory:
+  - name: /srv/node
```
Comment (fpytloun) [inline swift/container.sls]: Hard coded directory for some unknown purpose

File: swift/files/liberty/account-server.conf [hunk 7]
```diff
@@ -0,0 +1,37 @@
+{% from "swift/map.jinja" import account with context %}
+
+[DEFAULT]
+bind_ip = {{ account.bind.address }}
+bind_port = {{ account.bind.port }}
+# bind_timeout = 30
+# backlog = 4096
+user = swift
+swift_dir = /etc/swift
+devices = /srv/node
```
Comment (fpytloun) [inline swift/files/liberty/account-server.conf]: hard-coded

File: swift/files/liberty/container-server.conf [hunk 8]
```diff
@@ -0,0 +1,45 @@
+{% from "swift/map.jinja" import container with context %}
+
+[DEFAULT]
+bind_ip = {{ container.bind.address }}
+bind_port = {{ container.bind.port }}
+# bind_timeout = 30
+# backlog = 4096
+user = swift
+swift_dir = /etc/swift
+devices = /srv/node
```
Comment (fpytloun) [inline swift/files/liberty/container-server.conf]: hard-coded

File: swift/files/liberty/dispersion.conf [hunk 9]
```diff
@@ -0,0 +1,44 @@
+[dispersion] 
+{%- if pillar.swift.common.dispersion is defined %}
+{%- for k, v in pillar.swift.common.dispersion.iteritems() %}
+{{ k }} = {{ v }}
```
Comment (fpytloun) [inline swift/files/liberty/dispersion.conf:5]: Code that allow to do define anything, this is not correct metadata-driven approach.

File: swift/files/liberty/memcache.conf [hunk 10]
```diff
@@ -0,0 +1,32 @@
+{% from "swift/map.jinja" import common with context %}
+
+[memcache]
+# You can use this single conf file instead of having memcache_servers set in
+# several other conf files under [filter:cache] for example. You can specify
+# multiple servers separated with commas, as in: 10.1.2.3:11211,10.1.2.4:11211
+# (IPv6 addresses must follow rfc3986 section-3.2.2, i.e. [::1]:11211)
+# memcache_servers = 127.0.0.1:11211
+memcache_servers = {%- for member in common.cache.members %}{{ member.host }}:{{ member.port }}{% if not loop.last %},{% endif %}{%- endfor %}
```
Comment (fpytloun) [inline swift/files/liberty/memcache.conf]: port should have memcache default

File: swift/files/liberty/object-server.conf [hunk 11]
```diff
@@ -0,0 +1,39 @@
+{% from "swift/map.jinja" import object with context %}
+
+[DEFAULT]
+bind_ip = {{ object.bind.address }}
+bind_port = {{ object.bind.port }}
+# bind_timeout = 30
+# backlog = 4096
+user = swift
+swift_dir = /etc/swift
+devices = /srv/node
```
Comment (fpytloun) [inline swift/files/liberty/object-server.conf]: hard-coded

File: swift/files/liberty/proxy-server.conf [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+
+[filter:list-endpoints]
+use = egg:swift#list_endpoints
+
+{%- if proxy.identity is defined %}
+[filter:authtoken]
+paste.filter_factory = keystonemiddleware.auth_token:filter_factory
+identity_uri = http://{{ proxy.identity.host }}:35357/
+auth_url = http://{{ proxy.identity.host }}:35357/
+auth_uri = http://{{ proxy.identity.host }}:5000/
```
Comment (fpytloun) [inline swift/files/liberty/proxy-server.conf:57]: auth_url and auth_uri seems to be duplicate, the same for identity_uri

File: swift/files/liberty/swift.conf [hunk 13]
```diff
@@ -0,0 +1,182 @@
...
+# storage-policy:0.
+#
+# A 'policy_type' argument is also supported but is not mandatory.  Default
+# policy type 'replication' is used when 'policy_type' is unspecified.
+{%- for ring in pillar.swift.ring_builder.rings %}
+{%- if ring.get("enabled", True) %}
+
+[storage-policy:{{ loop.index0 }}]
+name = {{ ring.get('policy', {}).get('name', "Policy-"~loop.index0) }}
+{%- for k, v in ring.get('policy', {}).iteritems() %}
```
Comment (fpytloun) [inline swift/files/liberty/swift.conf:34]: Same here

File: swift/init.sls [hunk 14]
```diff
@@ -18,3 +18,19 @@ include:
 {%- if pillar.swift.ring_builder is defined %}
 - swift.ring_builder
 {%- endif %}
+
+
+#swift_group_and_user:
```
Comment (fpytloun) [inline swift/init.sls]: Why? Also swift user/group should have fixed uid/gid in the same matter as eg. nova.

File: swift/object.sls [hunk 15]
```diff
@@ -6,6 +6,16 @@ swift_object_packages:
   pkg.installed:
   - names: {{ object.pkgs }}
 
+swift_object_node_directory:
+  file.directory:
+  - name: /srv/node
```
Comment (fpytloun) [inline swift/object.sls]: hard-coded

File: swift/files/liberty/proxy-server.conf (hunk 12 above)
Comment (epcim) [reply swift/files/liberty/proxy-server.conf:57]: yes - changed in swift, swift documentation is wrong. to keep backward compatible, working even doc's says the other I kept both.  approach is to keep both for liberty, remove one any higher.

File: swift/files/liberty/dispersion.conf (hunk 9 above)
Comment (epcim) [reply swift/files/liberty/dispersion.conf:5]: will fire an issue once merged and find junior to fix.

File: swift/files/liberty/swift.conf (hunk 13 above)
Comment (epcim) [reply swift/files/liberty/swift.conf:34]: was an intention here to keep formula flexible during initial evolution, subject of change in future versions. Will create an issue for that.

**Pull request merged**